sdk/lib/io/data_transformer.dart

```diff
@@ -526,20 +510,17 @@ class _FilterSink extends ByteConversionSink {
         _sink.add(out);
       }
     } catch (e) {
-      // TODO(kevmoo): not sure why this isn't a try/finally
```

Comment:
Or why the sink isn't closed if it throws.
Looks weird. Consider keeping the comment, until someone investigates.

Maybe it assumes that only the sink `add` can throw, but that's optimistic, and why is the filter operation inside the `try` then?

Reply:
Ok